### PR TITLE
Add traefik

### PR DESCRIPTION
**Pull request by ashish-rf**
### Description of the change

<Added Traefik Image to the repo>

### Applicable issues

  - fixes #

### Checklist

- [ ] Added to the `README.md` using [readme-generator](https://github.com/rapidfort/community-images/readme-generator)

### CODE REVIEW

File: community_images/common/orchestrator/main.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def __init__(self, args):
         self.docker_client = docker.from_env()
         self.publish = args.publish
         self.force_publish = args.force_publish
-        self.config_name = self.args.config
```
Comment (dreamer-nitj) [inline community_images/common/orchestrator/main.py:27]: Why is this required

File: community_images/common/tests/common_commands.sh [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
 
 set -x
 set -e
 
 # Add common commands here which should be present in all hardened images
 
-function run_commands()
+run_commands()
 {
-    local cmd="$1"
```
Comment (dreamer-nitj) [inline community_images/common/tests/common_commands.sh:10]: Why this change?

File: community_images/common/tests/common_commands.sh [hunk 3]
```diff
@@ -1,32 +1,30 @@
...
+    cmd="$1"
+    if command -v "$cmd" --help > /dev/null
     then
         "$cmd" --help
     fi
 }
 
 sleep 1
 
-declare -a command_array=(
```
Comment (dreamer-nitj) [inline community_images/common/tests/common_commands.sh:19]: Same here, why this change is required?

File: community_images/traefik/traefik/.rfignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+sbin/mkmntdirs
+var/
+etc/
+usr/
```
Comment (dreamer-nitj) [inline community_images/traefik/traefik/.rfignore:4]: Should we check if we need to skip entire directories? Can we try to include coverage so that it is not required?

File: community_images/traefik/traefik/dc_coverage.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+echo "Json params for docker compose coverage = $JSON"
+
+PROJECT_NAME=$(jq -r '.project_name' < "$JSON_PARAMS")
+
+NETWORK_NAME="${PROJECT_NAME}"_default
+# Testing load balencer 
+for i in {1..3};
+do
+    echo "Attempt $i"
+    docker run -d --net "$NETWORK_NAME" rapidfort/curl "-H Host:whoami.docker.localhost http://127.0.0.1"
```
Comment (dreamer-nitj) [inline community_images/traefik/traefik/dc_coverage.sh:19]: What's the reason to have these same commands 3 times?

File: community_images/traefik/traefik/docker_coverage.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (dreamer-nitj) [inline community_images/traefik/traefik/docker_coverage.sh:1]: This file can be removed since it's not used anywhere

File: community_images/traefik/traefik/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+version: '2'
+
+services:
+  reverse-proxy:
+    image: ${TRAEFIK_IMAGE_REPOSITORY}:${TRAEFIK_IMAGE_TAG}
+    # Enables the web UI and tells Traefik to listen to docker
+    command:                                                                                                            
+      #- "--log.level=DEBUG"
```
Comment (dreamer-nitj) [inline community_images/traefik/traefik/docker-compose.yml:8]: Remove this line

**Review by dreamer-nitj — COMMENTED** [review]
Few comments

File: README.md [hunk 8]
```diff
@@ -42,36 +42,36 @@ We’ve optimized and hardened some of the most popular container images on Dock
```
Comment (dreamer-nitj) [inline README.md:42]: We also need to make changes to builder.lst and image.lst files and run image generator script

File: community_images/common/tests/common_commands.sh (hunk 3 above)
Comment (ashish-rf) [reply community_images/common/tests/common_commands.sh:19]: When changing the bash script to shell, this is the only way we can define an array, as shell doesn't support array's natively.
we needed the shell as the alpine image used by traefik doesn't have bash

File: community_images/common/tests/common_commands.sh (hunk 2 above)
Comment (ashish-rf) [reply community_images/common/tests/common_commands.sh:10]: need to use POSIX shell instead of bash for traefik image

File: community_images/common/orchestrator/main.py (hunk 1 above)
Comment (ashish-rf) [reply community_images/common/orchestrator/main.py:27]: if you run the command

python3 common/orchestrator/main.py hourly_run traefik/traefik/

 from 
/rapidfort/community-images/community_images

then extra "/" at the end of traefik/traefik/ makes out the calculation of directory 3 level above instead of two , which results in file not found. the extra "/" is automatically added because directory selection was made by pressing tab.

File: community_images/traefik/traefik/.rfignore (hunk 4 above)
Comment (ashish-rf) [reply community_images/traefik/traefik/.rfignore:4]: Most of the files are alpine image specific. Keys and certs, with a couple of executables from alpine. As there was very less  reduction in size 3-4% , we agreed to include these, happy to have a look together on these directories.

File: community_images/traefik/traefik/dc_coverage.sh (hunk 5 above)
Comment (ashish-rf) [reply community_images/traefik/traefik/dc_coverage.sh:19]: fixed , needed only once

File: community_images/traefik/traefik/.rfignore (hunk 4 above)
Comment (codervinod) [reply community_images/traefik/traefik/.rfignore:4]: This is unacceptable. We cant whitelist entire root level directories

**Review by codervinod — CHANGES_REQUESTED** [review]

**Review by codervinod — APPROVED** [review]

File: community_images/common/tests/common_commands.sh (hunk 2 above)
Comment (ddooley77) [reply community_images/common/tests/common_commands.sh:10]: does it break anything else?
